src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -17,14 +17,53 @@ public Rectangle(Integer width, Integer length, String color) {
     }
 
     public Integer getWidth() {
+
         return width;
     }
 
     public Integer getLength() {
+
         return length;
     }
 
     public String getColor() {
+
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) obj;
+        if (obj.getClass() != this.getClass()) {
+            return false;
+        }
+        if (width == null && rectangle.width == null && length == null && rectangle.length == null
+                && color == null && rectangle.color == null) {
+            return true;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+
+            return width.equals(rectangle.width) && length.equals(rectangle.length)
+                    && color.equals(rectangle.color);
+        }
```

Comment:
see checklist: While comparing fields at equals, use single return statement for all fields(e.g. return check1 && check2 && check3;)